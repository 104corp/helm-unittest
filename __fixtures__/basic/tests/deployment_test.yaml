suite: test deployment
templates:
  - deployment.yaml
tests:
  - it: should pass all assertions
    values:
      - ./values/image.yaml
    set:
      a.b.c: ABC
      x.y.z: XYZ
    asserts:
      - equal:
          path: spec.template.spec.containers[0].image
          value: apache:latest
      - notEqual:
          path: spec.template.spec.containers[0].image
          value: nginx:stable
      - matchRegex:
          path: spec.template.spec.containers[0].image
          pattern: apache.*
      - notMatchRegex:
          path: spec.template.spec.containers[0].image
          pattern: nginx.*
      - contains:
          path: spec.template.spec.containers[0].ports
          content:
            containerPort: 80
      - notContains:
          path: spec.template.spec.containers[0].ports
          content:
            containerPort: 8080
      - isNull:
          path: spec.template.nodeSelector
      - isNotNull:
          path: spec.template
      - isEmpty:
          path: spec.template.spec.containers[0].resources
      - isNotEmpty:
          path: spec.template.spec.containers[0]
      - isKind:
          of: Deployment
      - isAPIVersion:
          of: extensions/v1beta1
      - hasDocuments:
          count: 1
      - matchSnapshot: {}
