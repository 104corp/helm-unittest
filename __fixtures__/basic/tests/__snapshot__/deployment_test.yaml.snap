should pass all assertions:
  1: |
    apiVersion: extensions/v1beta1
    kind: Deployment
    metadata:
      labels:
        app: basic
        chart: basic-0.1.0
        heritage: Tiller
        release: RELEASE-NAME
      name: RELEASE-NAME-basic
    spec:
      replicas: 1
      template:
        metadata:
          labels:
            app: basic
            release: RELEASE-NAME
        spec:
          containers:
          - image: apache:latest
            imagePullPolicy: Always
            livenessProbe:
              httpGet:
                path: /
                port: 80
            name: basic
            ports:
            - containerPort: 80
            readinessProbe:
              httpGet:
                path: /
                port: 80
            resources: {}
